diagnosis/getkuberneteslogs.sh

```diff
@@ -2,160 +2,194 @@
+        y|Y|yes|YES ) echo "Continue";;
+        n|N|no|NO) echo "Aborting"; exit 0 ;;
+        * ) echo "Invalid option '$choice'. Stopping the log collection process"; exit 0;;
         esac
     fi
 
-    IDENTITYFILEBACKUPPATH="/home/$AZUREUSER/IDENTITYFILEBACKUP"
-
-    # Remove existing scrit if ther is
-    ssh -t -i $IDENTITYFILE $AZUREUSER@$HOST "if [ -f /home/$AZUREUSER/collectlogsmanager.sh ]; then sudo rm -f /home/$AZUREUSER/collectlogsmanager.sh; fi;"
-    ssh -t -i $IDENTITYFILE $AZUREUSER@$HOST "if [ -f /home/$AZUREUSER/collectlogs.sh ]; then sudo rm -f /home/$AZUREUSER/collectlogs.sh; fi;"
-
-    #Backup id_rsa
-    ssh -t -i $IDENTITYFILE $AZUREUSER@$HOST "if [ -f /home/$AZUREUSER/.ssh/id_rsa ]; then mkdir -p $IDENTITYFILEBACKUPPATH;  sudo mv /home/$AZUREUSER/.ssh/id_rsa $IDENTITYFILEBACKUPPATH; fi;"
-
-    #Copy id_rsa into Kubernete Host VM
-    scp -i $IDENTITYFILE $IDENTITYFILE $AZUREUSER@$HOST:/home/$AZUREUSER/.ssh/id_rsa
-
-    #set up permission and  Download the script
-    ssh -t -i $IDENTITYFILE $AZUREUSER@$HOST "if [ -f /home/$AZUREUSER/.ssh/id_rsa ]; then sudo chmod 400 /home/$AZUREUSER/.ssh/id_rsa; cd /home/$AZUREUSER; curl -O https://raw.githubusercontent.com/msazurestackworkloads/azurestack-gallery/master/diagnosis/collectlogsmanager.sh; curl -O https://raw.githubusercontent.com/msazurestackworkloads/azurestack-gallery/master/diagnosis/collectlogs.sh ;sudo chmod 744 collectlogsmanager.sh;  fi;"
-
-    ssh -t -i $IDENTITYFILE $AZUREUSER@$HOST "cd /home/$AZUREUSER; ./collectlogsmanager.sh;"
-
-    #Copy logs back to local machine
-    scp -r -i $IDENTITYFILE $AZUREUSER@$HOST:/home/$AZUREUSER/kubernetesalllogs $LOGFILEFOLDER
-
-    #Restore id_rsa
-    ssh -t -i $IDENTITYFILE $AZUREUSER@$HOST "if [ -f $IDENTITYFILEBACKUPPATH/id_rsa ] ; then sudo mv $IDENTITYFILEBACKUPPATH/id_rsa /home/$AZUREUSER/.ssh/id_rsa; sudo rm -r -f $IDENTITYFILEBACKUPPATH; fi;"
-
-    # Delete scrits
-    ssh -t -i $IDENTITYFILE $AZUREUSER@$HOST "if [ -f /home/$AZUREUSER/collectlogsmanager.sh ]; then sudo rm -f /home/$AZUREUSER/collectlogsmanager.sh; fi;"
-    ssh -t -i $IDENTITYFILE $AZUREUSER@$HOST "if [ -f /home/$AZUREUSER/collectlogs.sh ]; then sudo rm -f /home/$AZUREUSER/collectlogs.sh; fi;"
+    echo "[$(date +%Y%m%d%H%M%S)][INFO] About to collect cluster logs"
+
+    # Backup id_rsa
+    IDENTITYFILEBACKUP="~/.ssh/bak/id_rsa"
+    ssh -q -i $IDENTITYFILE $AZUREUSER@$HOST "if [ -f .ssh/id_rsa ]; then mkdir -p $IDENTITYFILEBACKUP; cp .ssh/id_rsa $IDENTITYFILEBACKUP; fi;"
+
+    # Copy id_rsa into Kubernete Host VM
+    echo "[$(date +%Y%m%d%H%M%S)][INFO] Uploading private key"
+    scp -q -i $IDENTITYFILE ~/.ssh/id_rsa $AZUREUSER@$HOST:/home/$AZUREUSER/.ssh/id_rsa
+    ssh -tq -i $IDENTITYFILE $AZUREUSER@$HOST "if [ -f .ssh/id_rsa ]; then sudo chmod 400 .ssh/id_rsa; fi;"
+
+    echo "[$(date +%Y%m%d%H%M%S)][INFO] Uploading scripts"
+    # scp -q -i $IDENTITYFILE common.sh $AZUREUSER@$HOST:/home/$AZUREUSER/common.sh
```

Comment:
As discussed please put th ecomment that this is in case user wants to run manually and not downloading the file from github.